docs/contributing.md

```diff
@@ -0,0 +1,134 @@
+# Contributing to UniFFI
+
+Anyone is welcome to help with the UniFFI project. Feel free to get in touch with other community members on Matrix or through issues here on GitHub.
+
+* Matrix: [#rust-components:mozilla.org](https://chat.mozilla.org/#/room/#rust-components:mozilla.org)
+* The [issue list](https://github.com/mozilla/uniffi-rs/issues)
+
+Participation in this project is governed by the
+[Mozilla Community Participation Guidelines](https://www.mozilla.org/en-US/about/governance/policies/participation/).
+
+## Building the project
+
+You will need a working [Rust](https://www.rust-lang.org/) development environment in order to build this project.
+We recommend using [`rustup`](https://rustup.rs/) to install the latest Rust toolchain and stay up-to-date with releases.
+
+In the root of your repository checkout, run:
+
+```
+cargo build
+```
+
+If you have `rustup` installed then this should work without errors; please [file an issue](https://github.com/mozilla/uniffi-rs/issues)
+if it doesn't work for you.
+
+## Running the tests
+
+In the root of your repository checkout, run:
+
+```
+cargo test
+```
+
+This should run, but is very likely to report test failures.
+
+Since this project generates API bindings for multiple languages, running the full test suite requires
+a working host environment for multiple languages.
+
+We publish a Docker image with the necessary dependencies pre-installed, and you can run `cargo test`
+from inside this Docker image using a helper script like so:
+
+```
+./docker/cargo-docker.sh test
+```
+
+This is convenient, but slower than running `cargo` locally. For rapid testing and iteration we
+recommend installing the language dependencies on your local machine. In order to locally run the full
+test suite you will need:
+
+* Kotlin:
+  * `kotlinc`, the [Kotlin command-line compiler](https://kotlinlang.org/docs/command-line.html)
+  * The [Java Native Access](https://github.com/java-native-access/jna#download) JAR downloaded and its path
+    added to your `$CLASSPATH` environment variable.
+* Swift:
+  * `swift` and `swiftc`, the [Swift command-line tools](https://swift.org/download/).
+  * The Swift `Foundation` package.
+* Python:
+  * A `python3` interpreter.
+* Ruby:
+  * A `ruby` interpreter.
+  * The [`FFI`](https://github.com/ffi/ffi) Ruby gem, installable via `gem install FFI`.
+
+
+## Navigating the code
```

Comment:
Very nice section! This will definitely make it easier even when coming back to uniffi after a brief hiatus!